### PR TITLE
Add deprecation warning to sox backend

### PR DESCRIPTION
**Pull request by mthrok**
See #903

### CODE REVIEW

File: torchaudio/backend/utils.py [hunk 1]
```diff
@@ -49,6 +49,10 @@ def set_audio_backend(backend: Optional[str]):
     if backend is None:
         module = no_backend
     elif backend == 'sox':
+        warnings.warn(
+            '"sox" backend is deprecated. '
+            'The default backend will be changeed to "sox_io" backend in 0.8.0 and '
```
Comment (vincentqb) [inline torchaudio/backend/utils.py:54]: nit: "changed"

File: torchaudio/backend/utils.py [hunk 2]
```diff
@@ -49,6 +49,10 @@ def set_audio_backend(backend: Optional[str]):
...
+            '"sox" backend will be removed in 0.9.0. Please migrate to "sox_io" backend.')
```
Comment (vincentqb) [inline torchaudio/backend/utils.py:55]: The instruction to suppress the warning should be more explicit. A link to the instructions or the command?

File: torchaudio/backend/utils.py [hunk 3]
```diff
@@ -49,6 +49,10 @@ def set_audio_backend(backend: Optional[str]):
     if backend is None:
         module = no_backend
     elif backend == 'sox':
+        warnings.warn(
+            '"sox" backend is deprecated. '
```
Comment (vincentqb) [inline torchaudio/backend/utils.py:53]: nit: I'd say "is being deprecated", though both seems fine :)

**Review by vincentqb — APPROVED** [review]
LGTM

File: torchaudio/backend/utils.py (hunk 3 above)
Comment (mthrok) [reply torchaudio/backend/utils.py]: changed it to `being deprecated`

File: torchaudio/backend/utils.py (hunk 2 above)
Comment (mthrok) [reply torchaudio/backend/utils.py]: Added the link to the issue.

File: torchaudio/backend/utils.py (hunk 1 above)
Comment (mthrok) [reply torchaudio/backend/utils.py]: fixed typo.

**Pull request merged**